packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -32,24 +32,28 @@ enum Orientation {
 /// exception, unless the `nullOk` argument is set to true, in which case it
 /// returns null.
 ///
-/// MediaQueryData includes two [EdgeInsets] values:
-/// [padding] and [viewInsets]. These
-/// values reflect the configuration of the device and are used by
-/// many top level widgets, like [SafeArea] and the Cupertino and
-/// Material scaffold widgets. The padding value defines areas that
-/// might not be completely visible, like the display "notch" on the
-/// iPhone X. The viewInsets value defines areas that aren't visible at
-/// all, typically because they're obscured by the device's keyboard.
+/// MediaQueryData includes three [EdgeInsets] values:
+/// [padding], [viewPadding], and [viewInsets]. These values reflect the
+/// configuration of the device and are used by many top level widgets, like
+/// [SafeArea] and the Cupertino and Material [Scaffold] widgets. The padding
+/// value defines areas that might not be completely visible, like the display
+/// "notch" on the iPhone X. The viewInsets value defines areas that aren't
+/// visible at all, typically because they're obscured by the device's keyboard.
+/// Similar to viewInsets, viewPadding does not differentiate padding in areas
+/// that may be obscured. For example, by using the viewPadding property,
+/// padding would defer to the iPhone "safe area" regardless of whether a
+/// keyboard is showing.
 ///
-/// The viewInsets and padding values are independent, they're both
+/// The viewInsets, viewPadding, and padding values are independent, they're
 /// measured from the edges of the MediaQuery widget's bounds. The
 /// bounds of the top level MediaQuery created by [WidgetsApp] are the
 /// same as the window that contains the app.
 ///
-/// Widgets whose layouts consume space defined by [viewInsets] or
-/// [padding] should enclose their children in secondary MediaQuery
+/// Widgets whose layouts consume space defined by [viewInsets], [viewPadding],
+/// or [padding] should enclose their children in secondary MediaQuery
 /// widgets that reduce those properties by the same amount.
-/// The [removePadding] and [removeInsets] methods are useful for this.
+/// The [removePadding], [removeViewPadding], and [removeInsets] methods are
+/// useful for this.
```

Comment:
I think we might want a See also section here linking to some relevant properties on [ui.window]

Reply:
I added references to `ui.window`